### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by t-bello7**
- Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
- `On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -95,12 +114,46 @@ def create_rental
...
+      when 'Teacher'
+        new_arr << { id: p.id, name: p.name, age: p.age, person_type: p.class.name, specialization: p.specialization }
+      when 'Student'
+        new_arr << { id: p.id, name: p.name, classroom: p.classroom, age: p.age,
+                     parent_permission: p.parent_permission, person_type: p.class.name }
+      end
+    end
+    File.write('files/persons.json', JSON.generate(new_arr)) if new_arr.length.positive?
+  end
 end
```
Comment (FrederickMih) [inline app.rb]: - [ ] On exit from the app, I got this console error on the screenshot below:

Screenshot from 2022-11-09 16-09-01

- Based on the requirements of this project all the data registered(people, books, rentals) should be saved into JSON files. Kindly fix the issue.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @t-bello7, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 


- Correct Gitflow has been followed :heavy_check_mark: 
- PR has a good title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 


## Required Changes ♻️

- [ ] Hi @team, this is a collaborative project. The Authors section of the README.md file should contain all members who contributed to this project.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 2]
```diff
@@ -1,15 +1,22 @@
 require_relative './app'
 require_relative './modules/choose_operation'
 require_relative './modules/display'
+require_relative './modules/load_data'
 
 @app = App.new
 
 def start_program
+  load_books(@app)
+  load_rentals(@app)
```
Comment (thecodechaser) [inline main.rb]: - [ ] After creating rentals and exiting the app, When I run the app again I get get the below error. Please fix this issue and make sure that rentals are loaded successfully after starting the app.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @milen-ale, and @t-bello7👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] As requested by the previous reviewer, Please make sure the Authors section of the README file should contain all members who contributed to this project.

- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi Team👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**